### DISCUSSION TITLE
[V5] How to define pagination meta keys with lucid query builder

### DISCUSSION THREAD

**Discussion by 3rownz** [adonisjs/core#2638] [Help]
Im trying to set the pagination keys using the new `paginationMetaKeys()`. I have followed the documentation and tried defining them on the controller and then tried globally on a provider. 

- Is there no way to use the query builder with the new pagaination changes? 
- I would also like to omit certain unused keys from the pagination, there no way to do this?

```
//I want to use it like this, but no changes to output
const products = Product.query().select(*).paginate(page, limit) 

//This is the only way that works
const products = await Database.from('products').paginate(page, limit)

**Comment by romileon** [comment]
still no answer?

**Reply by thetutlage** [reply]
Yup. Coz, the author did not share the code of what they have tried so far.
